lib/date_time_parser.ex

```diff
@@ -230,66 +239,37 @@ defmodule DateTimeParser do
     end
   end
 
-  defp do_datetime_parse(string) do
-    cond do
-      String.contains?(string, "/") ->
-        DateTimeParser.DateTime.parse_us(string)
-
-      epoch_regex_capture = Regex.named_captures(@epoch_regex, string) ->
-        Epoch.parse(epoch_regex_capture)
-
-      Regex.match?(@serial_regex, string) ->
-        Serial.parse(string)
-
-      true ->
-        DateTimeParser.DateTime.parse(string)
-    end
-  end
-
   @doc """
-  Parse `%Time{}` from a string.
+  Parse `%Time{}` from a string. Accepts options `t:parse_time_options/0`
   """
-  @spec parse_time(String.t() | nil) :: {:ok, Time.t()} | {:error, String.t()}
-  def parse_time(string) when is_binary(string) do
-    case string |> clean() |> do_time_parse() do
-      {:ok, tokens, _, _, _, _} ->
-        to_time(tokens)
-
+  @spec parse_time(String.t() | nil, parse_time_options()) ::
+          {:ok, Time.t()} | {:error, String.t()}
+  def parse_time(string, opts \\ [])
+
+  def parse_time(string, opts) when is_binary(string) do
+    with {:ok, parser} <- string |> clean() |> Parser.build(:time, opts),
+         {:ok, time} <- parser.mod.parse(parser) do
+      {:ok, time}
+    else
       _ ->
         {:error, "Could not parse #{string}"}
     end
   end
 
-  def parse_time(nil), do: {:error, "Could not parse nil"}
-  def parse_time(value), do: {:error, "Could not parse #{value}"}
+  def parse_time(nil, _opts), do: {:error, "Could not parse nil"}
+  def parse_time(value, _opts), do: {:error, "Could not parse #{value}"}
```

Comment:
This string interpolation will throw an error if it gets a value that doesn't implement the String.Chars protocl (e.g. tuples). 
It looks like the two previous function heads match on strings and nils.

![image](https://user-images.githubusercontent.com/10181938/73776387-a2f55200-4755-11ea-876f-2faf231e80f5.png)

It looks like this isn't a bug introduced by this PR and it might apply in several places so it might be worth addressing as a separate issue?

Reply:
thanks! good catch I should `#{inspect value}` instead. I'll open another PR to address this.